src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -165,6 +168,23 @@ export class CellExecution {
         );
     }
 
+    public async interrupt(timeoutMs: number): Promise<InterruptResult> {
+        // Go through our active executions and interrupt each one.
+        const results = await Promise.all(
```

Comment:
We might want to ensure this method cannot be called more than once & if already completed.
I.e. `if this._interrupted || this._completed) { return false}`
Where ``this._interrupted` is a promise that contains the result for the previous `interrupt` call for this cell execution (i.e. all interrupts will be doing the same thing - pointing to the same promise)

Reply:
& once the promise resolves we might want to clear that propery.
This way if user clicks multiple times, they will get the samae interrupt promise.
After user gets the interrupt error message if they try again, it should start a new interrupt sequence (without using the same - previously resolved promise).

Reply:
Good idea.